test/extended/router/http2.go

```diff
@@ -172,25 +255,123 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 						return false, nil // could be 503 if service not ready
 					}
 					if tc.statusCode == 0 {
+						resp.Body.Close()
 						return false, nil
 					}
 					if resp.StatusCode != tc.statusCode {
+						resp.Body.Close()
 						e2e.Logf("[test #%d/%d]: config: %s, expected status: %v, actual status: %v", i+1, len(testCases), testConfig, tc.statusCode, resp.StatusCode)
+						return false, nil
 					}
-					return resp.StatusCode == tc.statusCode, nil
+					return true, nil
 				})).NotTo(o.HaveOccurred())
 
 				if tc.expectedGetError != "" {
 					continue
 				}
 
+				o.Expect(resp).ToNot(o.BeNil(), testConfig)
 				o.Expect(resp.StatusCode).To(o.Equal(tc.statusCode), testConfig)
 				o.Expect(resp.Proto).To(o.Equal(tc.frontendProto), testConfig)
 				body, err := ioutil.ReadAll(resp.Body)
 				o.Expect(err).NotTo(o.HaveOccurred(), testConfig)
+				resp.Body.Close()
 				o.Expect(string(body)).To(o.Equal(tc.backendProto), testConfig)
 				o.Expect(resp.Body.Close()).NotTo(o.HaveOccurred())
```

Comment:
This seems odd to check, but if you need to, then disregard my other comments about `resp.Body.Close`

Reply:
Yeah, there should only have been one call - and this one was intentional.